### ISSUE TITLE
Configuration loading

### ISSUE THREAD

**Issue by weirdan** [weirdan/doctrine-psalm-plugin#10]
This plugin needs to be able to load doctrine configuration for functionality like checking entity types, findBy* methods, etc.

**Comment by vladyslavstartsev** [comment]
@ weirdan Can you please describe in more details? 
I would like to try my self to fix\resolve this

**Comment by vladyslavstartsev** [comment]
@weirdan some why you have not been mentioned

**Comment by iluuu1994** [comment]
[phpstan-doctrine](https://github.com/phpstan/phpstan-doctrine#configuration) allows providing an `EntityManager` instance so that the plugin can infer repositories. Maybe that would be easier?

**Comment by vladyslavstartsev** [comment]
> stubbing the connection so that no database is required
> replacing reflection service so that classes need not to be loaded

@iluuu1994  I think those comments mean that @weirdan  doesn't want to load whole application in order to get entity manager.

But on the other hand, I'm more on the @iluuu1994 side, only the real entity manager can tell us real return types. And it will also be able to parse and validate DQL.

On the other hand, it will be dynamic code analysis ( nothing bad, but its not SA anymore)

@weirdan can you please explain your POV

**Comment by iluuu1994** [comment]
@vladyslavstartsev 

>  I think those comments mean that @weirdan doesn't want to load whole application in order to get entity manager.

Yeah but it could also be opt-in. If you want more accurate analysis you'd need to provide a real `EntityManager` instance. Otherwise we'd either have to:

1. Repeat the whole configuration in a psalm config
2. Make psalm understand every type of doctrine configuration (xml, yaml, annotations, etc) which is just not practical

**Comment by beberlei** [comment]
Doctrine has a static metadata driver / reflection component where you can read the configuration *almost* without executing code, only reading the configuration. You can look at the EntityGenerator or Convert Database to Entity commands to see how to use them. If you go low level enough, you don't need the EntityManager as far as i remember.

Another option that the Symfony psalm plugin uses is to load the "cached" metadata (or cahced container.xml in that case). Doctrine writes configuration to the cache as serialized class metadata. You could load that without needing much of anything runtime doctrine code.

**Comment by vladyslavstartsev** [comment]
I should probably mention it here
https://twitter.com/IssecMura/status/1274068540950302720?s=19

**Comment by michnovka** [comment]
Hi @weirdan just wanted to ask whether you have some spare time for this in the near future. Thanks!

**Comment by weirdan** [comment]
Unlikely to happen anytime soon.
